### PR TITLE
feat: split out api references from the view

### PR DESCRIPTION
**Pull request by amritk**
Not quite done but up for comment.

@geoffgscott is this what you were thinking, did you want the editor slotted out this far? OR maybe its slotted inside `ApiReferences` and we just use a boolean to import it or not. 
@hanspagel need to run this by you as well. Also I need to add the editor a few more places where it is used. I think we might need one more wrapper that has the editor inside of it.

### CODE REVIEW

File: projects/web/src/pages/ApiReferencePage.vue [hunk 1]
```diff
@@ -38,5 +45,9 @@ const configuration = reactive<ReferenceConfiguration>({
   </div> -->
   <ApiReference
     :configuration="configuration"
-    @changeTheme="(theme: ThemeId) => (configuration.theme = theme)" />
+    @changeTheme="(theme: ThemeId) => (configuration.theme = theme)">
+    <template #editor="slotProps">
+      <LazyLoadedSwaggerEditor v-bind="slotProps" />
```
Comment (amritk) [inline projects/web/src/pages/ApiReferencePage.vue:50]: Yep think I went too far here, will pull this back in under a boolean prop

**Comment by hanspagel** [comment]
Converted to draft while it’s work in progress. 🚧

**Comment by hanspagel** [comment]
This will probably have a lot of conflicts with #398. I’ll take a closer look once #398 is in (and resolve conflicts), but I think this could be the right direction.

**Comment by amritk** [comment]
Sounds good @hanspagel let me know if you need any help merging.

File: projects/web/src/pages/ApiReferencePage.vue [hunk 2]
```diff
@@ -38,5 +38,6 @@ const configuration = reactive<ReferenceConfiguration>({
   </div> -->
   <ApiReference
     :configuration="configuration"
-    @changeTheme="(theme: ThemeId) => (configuration.theme = theme)" />
+    @changeTheme="(theme: ThemeId) => (configuration.theme = theme)">
```
Comment (amritk) [inline projects/web/src/pages/ApiReferencePage.vue:41]:
```suggestion
    :hasEditor="false"
    @changeTheme="(theme: ThemeId) => (configuration.theme = theme)">
```
Try passing `hasEditor`  = false here to see it with no editor.

File: packages/api-reference/src/components/ApiReference.vue [hunk 3]
```diff
@@ -229,91 +222,36 @@ watch(
...
+  <ApiReferenceLayout
+    :currentConfiguration="currentConfiguration"
+    :isDarkMode="isDark"
+    :parsedSpec="parsedSpec"
+    :rawSpec="rawSpecRef"
+    :swaggerEditorRef="swaggerEditorRef"
+    @changeTheme="$emit('changeTheme', $event)"
+    @toggleDarkMode="toggleDarkMode">
+    <template #header>
+      <slot name="header" />
```
Comment (geoffgscott) [inline packages/api-reference/src/components/ApiReference.vue:244]: I don't think we need a slot here (@hanspagel what do you think?) but whether we have a slot or not the content should be provided here (either as slot fallback or as direct slot content) same as it was before.

File: packages/api-reference/src/components/ApiReference.vue [hunk 4]
```diff
@@ -325,309 +263,10 @@ function handleAIWriter(
...
-              @changeExample="swaggerEditorRef?.handleChangeExample"
-              @changeTheme="$emit('changeTheme', $event)"
-              @openSwaggerEditor="swaggerEditorRef?.handleOpenSwaggerEditor" />
-          </template>
-        </Content>
-      </div>
-      <div class="references-footer">
-        <slot name="footer" />
-      </div>
+        @startAIWriter="handleAIWriter" />}
```
Comment (geoffgscott) [inline packages/api-reference/src/components/ApiReference.vue:266]:
```suggestion
        @startAIWriter="handleAIWriter" />
```

File: packages/api-reference/src/components/ApiReference.vue (hunk 3 above)
Comment (amritk) [reply packages/api-reference/src/components/ApiReference.vue:244]: I believe this was just copied over from how it was done before 
image

File: packages/api-reference/src/components/ApiReference.vue (hunk 4 above)
Comment (amritk) [reply packages/api-reference/src/components/ApiReference.vue:266]: strange lol

**Review by hanspagel — APPROVED** [review]
successfully untangled 🫡

**Review by marclave — APPROVED** [review]
huge improvement!

**Pull request merged**